pkg/dbmate/db.go

```diff
@@ -202,6 +218,23 @@ func doTransaction(db *sql.DB, txFunc func(Transaction) error) error {
 	return tx.Commit()
 }
 
+func transactAndRollback(db *sql.DB, txFunc func(Transaction) error) error {
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+
+	if err := txFunc(tx); err != nil {
+		if err1 := tx.Rollback(); err1 != nil {
```

Comment:
There's nothing in the function signature for ```txFunc``` that tells me the relationship between getting an ```error``` back and the transaction state (did it ```Commit```? Who knows)...Seems like this is just easier to inline where you need it to avoid this

Reply:
Yea, your other suggestions for `transactAndRollback` also makes sense for `doTransaction`. I will make this change.